### PR TITLE
LRU Gc for memory

### PR DESCRIPTION
**Pull request by wu-hui**
see: https://github.com/firebase/firebase-js-sdk/pull/6943 and go/firestore-memory-lru

### CODE REVIEW

**Review by tom-andersen — COMMENTED** [review]
Looks good! Though I really don't like that production code has code to accommodate testing. I tried to add some comments to help minimize this.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/MemoryEagerGcSettings.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    private Builder() {}
+
+    @NonNull
+    public MemoryEagerGcSettings build() {
+      return new MemoryEagerGcSettings();
+    }
+  }
+
+  @Override
+  public int hashCode() {
```
[the remark this answers is not in the document]
Comment (wu-hui) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/MemoryEagerGcSettings.java:33]: Removed.

File: firebase-firestore/src/test/java/com/google/firebase/firestore/spec/SpecTestCase.java [hunk 2]
```diff
@@ -317,10 +324,19 @@ private void initClient() {
             maxConcurrentLimboResolutions,
             new FirebaseFirestoreSettings.Builder().build());
 
-    ComponentProvider provider =
-        initializeComponentProvider(configuration, garbageCollectionEnabled);
+    ComponentProvider provider = initializeComponentProvider(configuration, useEagerGcForMemory);
     localPersistence = provider.getPersistence();
+    if (localPersistence.getReferenceDelegate() instanceof SQLiteLruReferenceDelegate) {
```
[the remark this answers is not in the document]
Comment (wu-hui) [reply firebase-firestore/src/test/java/com/google/firebase/firestore/spec/SpecTestCase.java:329]: Fixed, thanks!

File: firebase-firestore/src/main/java/com/google/firebase/firestore/MemoryEagerGcSettings.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tom-andersen) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/MemoryEagerGcSettings.java:33]: I guess it was required?

**Review by tom-andersen — APPROVED** [review]
LGTM

File: firebase-firestore/src/main/java/com/google/firebase/firestore/MemoryEagerGcSettings.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (wu-hui) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/MemoryEagerGcSettings.java:33]: Oh yes.

**Pull request merged**